Source/cryptography/ICryptography.h

```diff
@@ -103,6 +106,35 @@ namespace Cryptography {
         virtual uint32_t Derive(const uint32_t privateKey, const uint32_t peerPublicKeyId, uint32_t& secretId /* @out */) = 0;
     };
 
+    struct IPersistent : virtual public Core::IUnknown {
+
+    enum { ID = ID_PERSISTENT };
+
+    enum keytype {
+        AES128,
+        AES256,
+        HMAC128,
+        HMAC160,
+        HMAC256
+    };
+
+    virtual ~IPersistent() { }
+
+    //Check if a named key exists in peristent storage
+    virtual uint32_t Exists(const string& locator, bool& result) const =0;
```

Comment:
@sebaszm I think you should add /* @out */ on the bool& result so -> 
virtual uint32_t Exists(const string& locator, bool& result /* @out */ ) const =0;
So that the ProxySTub generator can optimize the Proxy/stubs on not marshalling this boolean on the way in.. @sebaszm to confirm.
Do this for all parameters that are just intended to go out.

Reply:
Correct, all output-only parameters should have the @out tag.